src/GravityFormsExtensions.php

```diff
@@ -339,4 +340,127 @@ public function p4_gf_enforce_ajax(array $form_args): array
 
         return $form_args;
     }
+
+    /**
+     * Purges the caches of pages that contain a form.
+     *
+     * @param array $form The form to look for when purging page caches
+     * @param bool $is_new Form is new
+     *
+     */
+    public function p4_gf_clear_page_caches(array $form, bool $is_new = false): void
+    {
+        if ($is_new) {
+            return;
+        }
+
+        /**
+         * Filter hook to change post types that are cleared from cache if they contain a form that has changed
+         *
+         * @param array $post_types Array of post types to consider for cache clearing after form update
+         */
+        $post_types = apply_filters('planet4_form_cache_purge_post_types', [ 'page', 'post', 'campaign' ]);
```

Comment:
Could you please also add a `p4_action` post type here. [reference](https://github.com/greenpeace/planet4-master-theme/blob/main/src/ActionPage.php#L12).

Reply:
Good point, done!